src/components/gcds-checkbox/gcds-checkbox.tsx

```diff
@@ -73,21 +66,130 @@ export class GcdsCheckbox {
       this.errorMessage = "";
     } else if (!this.hasError && this.errorMessage) {
       this.hasError = true;
+    } else if (this.errorMessage == "") {
+      this.hasError = false;
     }
   }
   /**
    * Hint displayed below the label.
    */
   @Prop({ reflect: true, mutable: false }) hint: string;
 
+  /**
+   * Array of validators
+   */
+  @Prop({ mutable: true }) validator: Array<string | ValidatorEntry | Validator<string>>;
+
+  @Watch('validator')
+  validateValidator() {
+    if (this.validator && !this.validateOn) {
+      this.validateOn = "blur";
+    }
+  }
+
+  /**
+  * Set event to call validator
+  */
+  @Prop({ mutable: true }) validateOn: 'blur' | 'submit' | 'other';
+
+  /**
+  * Event listener for gcds-fieldset errors
+  */
+  @Listen('gcdsGroupError', { target: 'body'})
+  gcdsGroupError(e) {
+    if (e.srcElement.contains(this.el) && elementGroupCheck(this.name)) {
+      this.hasError = true;
+      this.parentError = e.detail;
+    } else if (!elementGroupCheck(this.name)) {
+      this.hasError = false;
+      this.parentError = "";
+    }
+  }
+  @Listen('gcdsGroupErrorClear', { target: 'body'})
+  gcdsGroupErrorClear(e) {
+    if (e.srcElement.contains(this.el) && this.hasError) {
+      this.hasError = false;
+      this.parentError = "";
+    }
+  }
+
+  /**
+   * State to handle when errors are passed down to component
+   */
+  @State() parentError: string;
+
+  /**
+   * Specifies if the checkbox is invalid.
+   */
+  @State() hasError: boolean;
+  @Watch('hasError')
+  validateHasError() {
```

Comment:
Same comment as above, the function is shared with:
- gcds-checkbox
- gcds-radio
- gcds-file-uploader
- gcds-select